docs/pages/reference/helm-reference/teleport-kube-agent.mdx

```diff
@@ -804,6 +925,31 @@ Enables the creation of a Kubernetes persistent volume to hold Teleport instance
 
 [Kubernetes reference](https://kubernetes.io/docs/concepts/storage/persistent-volumes/)
 
+#### Chart upgrades
+
+If `storage.enabled` is `false` and you are upgrading a legacy
+`teleport-kube-agent` chart, the chart will launch a Teleport pod as a
+`Deployment` instead of a `StatefulSet`, disabling persistent storage for the
+Teleport pod. If `storage.enabled` is set to `true` during an upgrade, the chart
+will create a webhook that deletes any legacy `Deployment` and launches a
+`StatefulSet` instead.
+
+#### New chart installations
+
+New installations of `teleport-kube-agent` always launch the Teleport pod as a
+`StatefulSet`, and `storage.enabled` configures persistent storage for the pod.
```

Comment:
After T11, Storage is no longer used to store Teleport Identity even if storage is enabled.
Storage is useful if you want to store session recordings locally.